eureka/src/main/java/com/linecorp/armeria/common/eureka/InstanceInfo.java

```diff
@@ -43,6 +46,33 @@ public final class InstanceInfo {
 
     private static final Logger logger = LoggerFactory.getLogger(InstanceInfo.class);
 
+    private static final AttributeKey<InstanceInfo> INSTANCE_INFO = AttributeKey.valueOf(
+            InstanceInfo.class, "INSTANCE_INFO");
+
+    /**
+     * Returns the {@link InstanceInfo} associated with the given {@link Endpoint}.
+     *
+     * @param endpoint The {@link Endpoint} whose {@link InstanceInfo} is to be retrieved.
+     * @return The {@link InstanceInfo} associated with the specified {@link Endpoint}.
+     */
+    @Nullable
+    public static InstanceInfo instanceInfo(Endpoint endpoint) {
+        requireNonNull(endpoint, "endpoint");
+        return endpoint.attr(INSTANCE_INFO);
+    }
+
+    /**
+     * Sets the Eureka {@link InstanceInfo} parameter to the {@link Endpoint} as an attribute.
+     *
+     * @param endpoint The {@link Endpoint} to which the {@link InstanceInfo} will be set as an attribute.
+     * @return The same {@link Endpoint} passed as a parameter.
+     */
+    public static Endpoint setInstanceInfo(Endpoint endpoint, InstanceInfo instanceInfo) {
```

Comment:
Should we move this method to internal API? Generally, I don't think, users need this API. This setter may be only used in `EurekaEndpointGroup`.

Reply:
The original issue was to retrieve metadata from an `Endpoint`. Also, I think there's no reason to hide this API from public. If you want to put this API into `InstanceInfo`, we might consider adding `InstanceInfoUtil`.

Reply:
Understood. Let’s proceed that way.